### PR TITLE
GODRIVER-3248 Update windows CI build

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-3248

## Summary

- Change distro to `windows-vsCurrent-small`
- Only run windows with 8.0 server

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2393,21 +2393,28 @@ buildvariants:
 
   - matrix_name: "tests-36-with-zlib-support"
     tags: ["pullrequest"]
-    matrix_spec: { version: ["3.6"], os-ssl-32: ["windows-64-go-1-20", "rhel87-64-go-1-20"] }
+    matrix_spec: { version: ["3.6"], os-ssl-32: ["rhel87-64-go-1-20"] }
```
Comment (blink1073) [inline .evergreen/config.yml:2396]: I think we should create windows versions without the pullrequest tag instead of removing windows altogether.

Comment (prestonvasquez) [reply .evergreen/config.yml:2396]: This PR still proposes testing for windows on 8.0 on pullrequest. Are you suggesting we also account for the waterfall case:

```yaml
  - matrix_name: "tests-80-plus-windows-zlib-zstd-support"
    matrix_spec: { version: ["4.2", "4.4", "5.0", "6.0", "7.0", "8.0"], os-ssl-40: ["windows-64-go-1-20"] }
    display_name: "${version} ${os-ssl-40}"
    tasks:
      - name: ".test !.enterprise-auth !.snappy"
```

Comment (blink1073) [reply .evergreen/config.yml:2396]: I want to ensure that we're testing 3.6-7.0 with Windows on git push.

Comment (prestonvasquez) [reply .evergreen/config.yml:2396]: I've updated the PR to only run Windows builds for server version 8.0 on PRs. However, still run 3.6-7.0 when merging a PR.

**Pull request merged**